src/main/java/duckie/command/AddCommand.java

```diff
@@ -0,0 +1,40 @@
+package duckie.command;
+
+import duckie.Storage;
+import duckie.Ui;
+import duckie.exception.DuckieException;
+import duckie.task.Task;
+import duckie.task.TaskList;
+
+/**
+ * Command to add task into the TaskList
+ */
+public class AddCommand extends Command {
+    private Task task;
+
+    /**
+     * Instantiate the AddCommand
+     * @param task Task to be added
+     */
+    public AddCommand(Task task) {
+        this.task = task;
+    }
+
+    /**
+     * Execute the command to add task into TaskList
+     * @param tasks TaskList containing all the tasks
+     * @param ui Ui to interact with the users
+     * @param storage Storage to write to File
+     * @throws DuckieException
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DuckieException {
+        tasks.addTask(this.task);
+        try {
+            storage.saveToFile(tasks.getTaskList());
```

Comment:
It depends on your reasoning as well but this approach would probably have a greater run time than perhaps just storing all the lists in the end when the user inputs the "bye" command.